### PR TITLE
[Playwright] PEPPER-515 Fix and update failed Playwright tests

### PR DESCRIPTION
**Pull request by aweng98**
- Updated UI screenshots.
- Removed some duplicated code.
- Updated test due to UI changes.
- Skip one test due one bug.
- Improved some functions to avoid flaky tests.
 
All tests passed several times on Dev env.

Screenshot 2023-01-17 at 12 38 48 PM

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -646,6 +646,12 @@ jobs:
             echo "export ANGIO_USER_PASSWORD=$angioUserPassword" >> playwright-env/envvars
             echo "export ANGIO_USER_EMAIL=$angioUser" >> playwright-env/envvars
 
+            # PANCAN
+            export pancanUser=$(vault read --format=json secret/pepper/test/v1/e2e | jq -r ".data.users | .[] | select(.app==\"pancan\") | .userName")
+            export pancanUserPassword=$(vault read --format=json secret/pepper/test/v1/e2e | jq -r ".data.users | .[] | select(.app==\"pancan\") | .password")
+            echo "export PANCAN_USER_PASSWORD=$pancanUserPassword" >> playwright-env/envvars
+            echo "export PANCAN_USER_EMAIL=$pancanUser" >> playwright-env/envvars
+
```
Comment (aweng98) [inline .circleci/config.yml:654]: Needed for Pancan tests on CircleCI.

File: playwright-e2e/authentication/auth-dsm.ts [hunk 2]
```diff
@@ -16,6 +17,5 @@ export async function login(page: Page, opts: { email?: string; password?: strin
   }
   await page.goto(DSM_BASE_URL, { waitUntil: 'load' });
   await fillInEmailPassword(page, { email, password, waitForNavigation: false });
+  await waitForNoSpinner(page);
 }
-
-export { createAccountWithEmailAlias } from 'authentication/auth-base';
```
Comment (aweng98) [inline playwright-e2e/authentication/auth-dsm.ts:21]: Not needed because DSM does not have "Create Account" UI.

File: playwright-e2e/lib/widget/button.ts [hunk 3]
```diff
@@ -8,13 +8,20 @@ export default class Button extends WidgetBase {
...
     this.elementLocator = ddpTestID
         ? this.rootLocator.locator(`button[data-ddp-test="${ddpTestID}"]`)
-        : exactMatch
-            ? this.rootLocator.locator(`xpath=//button[.//text()[normalize-space()="${label}"]]`)
-            : this.rootLocator.locator(`xpath=//button[.//text()[contains(normalize-space(),"${label}")]]`);
+        : label
+            ? exactMatch
+                ? this.rootLocator.locator(`xpath=.//button[.//text()[normalize-space()="${label}"]]`)
+                : this.rootLocator.locator(`xpath=.//button[.//text()[contains(normalize-space(),"${label}")]]`)
+            : this.rootLocator.locator(`xpath=.//button`).first();
```
Comment (aweng98) [inline playwright-e2e/lib/widget/button.ts:24]: Handle special edge case: If `label` is undefined.

File: playwright-e2e/tests/pancan/enrollment/adult-self-enrollment.spec.ts [hunk 4]
```diff
@@ -62,42 +63,48 @@ test.describe('Enroll myself as adult', () => {
...
-    const surveCervicalCancerPage = new SurveyCervicalCancerPage(page);
-    await surveCervicalCancerPage.waitForReady();
-    await surveCervicalCancerPage.cervicalCancerDiagnosedDate('March', '2015');
-    await surveCervicalCancerPage.fillCancerBodyPlaces('Appendix');
-    await surveCervicalCancerPage.cancerFree().radioButton('Yes').locator('label').click();
-    await surveCervicalCancerPage.fillBodyPlacesEverHadCancer('Appendix');
-    await surveCervicalCancerPage.checkTreatmentsReceived('Radiation');
-    await surveCervicalCancerPage.medicationsList().fill('many others');
-    await surveCervicalCancerPage.submit();
+    const surveyCervicalCancerPage = new SurveyCervicalCancerPage(page);
```
Comment (aweng98) [inline playwright-e2e/tests/pancan/enrollment/adult-self-enrollment.spec.ts:81]: Fixed spelling errors.

File: playwright-e2e/authentication/auth-rgp.ts [hunk 5]
```diff
@@ -17,19 +16,4 @@ export async function login(page: Page, opts: { email?: string; password?: strin
...
-  if (password == null) {
-    throw Error('Invalid parameters: RGP user password is undefined or null.');
-  }
-
-  const emailAlias = generateEmailAlias(email);
-  await page.locator('input#signUpEmail').fill(emailAlias);
-  await page.locator('input#signUpPassword').fill(password);
-  await page.click('button#signUpSubmit');
-  return emailAlias;
-}
```
Comment (aweng98) [inline playwright-e2e/authentication/auth-rgp.ts:35]: Cuplicate code.

File: playwright-e2e/pages/singular/dashboard/my-dashboard-page.ts [hunk 6]
```diff
@@ -59,9 +62,26 @@ export default class MyDashboardPage extends SingularPage {
...
+    } = opts;
     await this.getEnrollMyChildButton().click();
+    const enrollMyChildPage = new EnrollMyChildPage(this.page);
+    await enrollMyChildPage.whoInChildFamilyHasVentricleHeartDefect().check(who);
+    await enrollMyChildPage.howOldIsYourChild().fill(age);
+    await enrollMyChildPage.whereDoesChildLive().select('Select Country').selectOption(country);
+    await enrollMyChildPage.state().selectOption(state);
+    if (cognitiveImpairment) {
+      await enrollMyChildPage.doesChildHaveCognitiveImpairment().check(cognitiveImpairment, { exactMatch: true });
+    }
```
Comment (aweng98) [inline playwright-e2e/pages/singular/dashboard/my-dashboard-page.ts:84]: Refactored `enrollMyChild()` function to remove duplicated code in test sripts.

File: playwright-e2e/tests/singular/enrollment/adult-dependent-enrollment.spec.ts [hunk 7]
```diff
@@ -33,10 +33,11 @@ test.describe('Enrol an adult dependent', () => {
     await expect(locator).toContainText(itemName);
   };
 
+  /** Test skip due to bug https://broadworkbench.atlassian.net/browse/PEPPER-485. Re-enable after bug fix. */
   /**
    * Test case: https://docs.google.com/document/d/1vaiSfsYeDzEHeK2XOVO3n_7I1W0Z94Kkqx_82w8-Vpc/edit#heading=h.6snot4x1e1uw
    */
-  test('can finish adult-dependent-enrollment @enrollment @singular', async ({ page, homePage }) => {
+  test.skip('can finish adult-dependent-enrollment @enrollment @singular', async ({ page, homePage }) => {
```
Comment (aweng98) [inline playwright-e2e/tests/singular/enrollment/adult-dependent-enrollment.spec.ts:40]: PEPPER-485 is a bug.

**Review by Rinchan20 — APPROVED** [review]

File: playwright-e2e/lib/widget/select.ts [hunk 8]
```diff
@@ -11,35 +11,69 @@ export default class Select extends WidgetBase {
...
-        await this.page
-          .locator(`#${ariaControlsId}[role="listbox"]`)
-          .locator(`mat-option .mat-option-text >> text="${option}"`)
-          .click();
+        if (!ariaControlsId) {
+          throw Error('ERROR: Cannot find attribute "aria-controls"');
+        }
+        /* eslint-disable no-case-declarations */
+        const dropdown = this.page.locator(`#${ariaControlsId}[role="listbox"]`);
+        const ariaMultiSelectable = await dropdown.getAttribute('aria-multiselectable');
```
Comment (aweng98) [inline playwright-e2e/lib/widget/select.ts:66]: Handling multi-select options Select and single option Select.

File: playwright-e2e/lib/widget/table.ts [hunk 9]
```diff
@@ -18,9 +18,9 @@ export default class Table {
         : cssClassAttribute
             ? `table${cssClassAttribute}, mat-table${cssClassAttribute}`
             : 'table, mat-table, [role="table"]';
-    this.headerCss = '[role="columnheader"], th[class]';
-    this.rowCss = '[role="row"]:not(mat-header-row), tbody tr';
-    this.cellCss = 'td, [role="cell"]';
+    this.headerCss = '[role="columnheader"]';
+    this.rowCss = '[role="row"]:not([mat-header-row]):not(mat-header-row), tbody tr';
```
Comment (aweng98) [inline playwright-e2e/lib/widget/table.ts:22]: `mat-header-row` could be an attribute or property for the html Select web element.

**Review by GiCharkviani — APPROVED** [review]
LGTM
